### PR TITLE
match jira id from github commit

### PR DESCRIPTION
**Pull request by joejstuart**
Leaving this as draft right now since there's more to add.

### CODE REVIEW

File: cmd/github.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+var commit, token, repository, jiraMatch string
+var jira bool
+
+func init() {
+	rootCmd.AddCommand(githubCmd)
+	githubCmd.Flags().StringVar(&commit, "commit", "", "Return the commit information")
+	githubCmd.Flags().BoolVar(&jira, "jira", false, "Parse Jira id from the commit message")
+	githubCmd.Flags().StringVar(&jiraMatch, "jiraMatch", "(?i)RedHat JIRA Issue: ([a-zA-Z]+-\\d+)", "Regex to match Jira id with")
```
Comment (lcarva) [inline cmd/github.go:68]: Can we find a more flexible default value? Is "RedHat Jira Issue:" a prefix widely used anywhere?

**Review by lcarva — COMMENTED** [review]
Overall, I don't necessarily agree with the approach of fetching the commit information from the GitHub API for two reasons:
1. This check should run on the build pipeline. The git clone is already there. Let's validate that instead.
2. I find it unlikely that we'll only ever check a single commit. We may have to fetch 10s or 100s of commits for each image being validated. Using the GitHub API is going to introduce significant latency.
3. This ties the implementation to GitHub which I guess it's ok at this point since other parts of hacbs are already tied to it.

File: cmd/github.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	githubCmd.Flags().StringVar(&token, "token", "", "The github api token to use")
+	githubCmd.Flags().StringVar(&repository, "repository", "", "The github repository. Format: organization/repository")
+}
+
+// getCommit connects to github and fetches a commit.
+// It returns a commit
+func getCommit() (git.Commit, error) {
```
[the remark this answers is not in the document]
Comment (lcarva) [reply cmd/github.go:75]: Actually, just remove the variables altogether and simply use `githubCmd.Flags().String` and `githubCmd.Flags().Bool`!

**Comment by joejstuart** [comment]
> Overall, I don't necessarily agree with the approach of fetching the commit information from the GitHub API:
> 
> 1. This check should run on the build pipeline. The git clone is already there. Let's validate that instead.
> 2. I find it unlikely that we'll only ever check a single commit. We may have to fetch 10s or 100s of commits for each image being validated. Using the GitHub API is going to introduce significant latency.
> 3. This ties the implementation to GitHub which I guess it's ok at this point since other parts of hacbs are already tied to it.

I went back and forth on this also. Ultimately I decided we would probably need to support the api and retrieving the commits from the cloned repo. I say this because I can see this being invoked independent of the build pipeline. Retrieving multiple commits might be an issue from the api (which I can't believe, but seems to be true).

File: pkg/git/client.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	"github.com/google/go-github/v42/github"
+	"golang.org/x/oauth2"
+)
+
+type GithubAuth struct {
+	Token string
+}
+
+type gitClient struct {
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply pkg/git/client.go:17]: I did use a lot of structs which may be overkill. I'll see if I can combine any of them. They all belong to different functions which is why I broke them up.

File: cmd/github.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+			if flagJira {
+				jiraId, err := commit.MatchJira(flagJiraMatch)
+				if err != nil {
+					fmt.Println(err.Error())
+					os.Exit(-1)
+				}
+				jiraJson, _ := json.Marshal(jiraId)
+				fmt.Printf("%s\n", jiraJson)
+			} else {
+				commitJson, _ := json.Marshal(commit)
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply cmd/github.go]: How strongly do you feel about this? I kind of like if you intentionally specify --jira, you get just the jira id(s) back.

File: cmd/github.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	Use:   "github",
+	Short: "Interact with the github api",
+	Long: `Interact with the github api. Supported commands are getting a commit
+	and parsing a jira id from a commit message`,
+	Run: func(cmd *cobra.Command, args []string) {
+		flagCommit, _ := cmd.Flags().GetString("commit")
+		flagJira, _ := cmd.Flags().GetBool("jira")
+		flagJiraMatch, _ := cmd.Flags().GetString("jiraMatch")
+
+		if flagCommit != "" {
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply cmd/github.go]: I made --commit and --repository required, so you get the usage if not specified.

**Comment by simonbaird** [comment]
While writing notes in [the m6 planning doc|https://docs.google.com/document/d/12thczSsHgmZM9XH_CqYlIiyWCAXePF4lCDJKl6swQEY/edit#] I realized I don't think this belongs in ec-cli. EC shouldn't become a general purpose CI checker, right?

**Pull request closed without merging**